src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.cs

```diff
@@ -202,9 +190,25 @@ protected override SyntaxNode GetCallSiteContainerFromOutermostMoveInVariable(Ca
                 return declStatement.Parent;
             }
 
+            private bool ShouldPutUnsafeModifier()
```

Comment:
this used to be on the strongly typed CSharpXXX types, but were only ever called from a single location (here).  so instead of needing all the generic type system wonkyness to pass in one of these subtypes, just to call special methods (not on the base type), i just moved the mtehods to the places that use them.  simple, easy.